src/fs.h

```diff
@@ -51,12 +51,26 @@ class path : public std::filesystem::path
     // Disallow std::string conversion method to avoid locale-dependent encoding on windows.
     std::string string() const = delete;
 
+    std::string u8string() const
+    {
+        const auto utf8_str{std::filesystem::path::u8string()};
+        // utf8_str might either be std::string (C++17) or std::u8string
+        // (C++20). Convert both to std::string. This method can be removed
+        // after switching to C++20.
+        return std::string{reinterpret_cast<const char*>(utf8_str.data()), utf8_str.size()};
```

Comment:
Should be able to just use `return std::string(utf8_str.begin(), utf8_str.end());` here? See http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1423r2.html